Testing: git push auto fixed files

lukelloyd1985 opened the pull request:
<!-- Please ensure your PR title is brief and descriptive for a good changelog entry -->
<!-- Link to issue if there is one -->
<!-- markdownlint-disable -->

<!-- markdownlint-restore -->

<!-- Describe what the changes are -->

## Proposed Changes

Testing git add, commit & push when there are fixed sources files

## Readiness Checklist

### Author/Contributor
- [ ] Add entry to the [CHANGELOG](https://github.com/oxsecurity/megalinter/blob/main/CHANGELOG.md) listing the change and linking to the corresponding issue (if appropriate)
- [ ] If documentation is needed for this change, has that been included in this pull request

### Reviewing Maintainer
- [ ] Label as `breaking` if this is a large fundamental change
- [ ] Label as either `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`, or `performance`

nvuillam commented on megalinter/reporters/UpdatedSourcesReporter.py:
```diff
@@ -63,6 +64,11 @@ def produce_report(self):
                 "Download it from artifacts then copy-paste it in your local repo to apply linters updates"
             )
+
+            repo = git.Repo(os.path.realpath(self.github_workspace))
```
Do that only if APPLY_FIXES = all , or matches the linter :)

- self.master will give you the Linter instance
- self.master.master will give you the MegaLinter instance
- so self.master.master.github_workspace will work :)

Also, check if you are on GitHub: if you are, don't make git operation, as they are already defined in the next GitHub Action :) ( if it works, we can deprecate it in a next major version)

lukelloyd1985 replied:
how to check the git platform being used?

nvuillam replied:
if GITHUB_REPOSITORY is set, you're on GitHub actions :)

lukelloyd1985 replied:
*think* I've catered for both of those now? 🤞

nvuillam commented on megalinter/reporters/UpdatedSourcesReporter.py:
```diff
@@ -63,6 +64,11 @@ def produce_report(self):
+            repo = git.Repo(os.path.realpath(self.github_workspace))
+            repo.git.add(update=True)
+            repo.index.commit("megalinter auto fixes")
+            repo.git.push
```
Also add try except, I don't want the whole MegaLinter to crash in case UpdatedSourcesReporter crashes, I prefer to just have a warning message ^^

lukelloyd1985 replied:
put a try except around which will hopefully give the correct git error message as I can't imagine this is going to work first attempt 😆

nvuillam commented on megalinter/reporters/UpdatedSourcesReporter.py:
```diff
@@ -63,6 +64,16 @@ def produce_report(self):
+            apply_fixes = config.get_list(self.request_id, "APPLY_FIXES", "none")
+            if apply_fixes:
+                try:
+                    repo = git.Repo(os.path.realpath(self.master.master.github_workspace))
```
sorry, master is MegaLinter instance, so it will be self.master.github_workspace :)

lukelloyd1985 replied:
switched 😄

nvuillam commented on megalinter/reporters/UpdatedSourcesReporter.py:
```diff
@@ -63,6 +64,16 @@ def produce_report(self):
                 "Download it from artifacts then copy-paste it in your local repo to apply linters updates"
             )
+
+            apply_fixes = config.get_list(self.request_id, "APPLY_FIXES", "none")
```
self.master.request_id :)

lukelloyd1985 replied:
I overlooked that 🤦‍♂️ good catch 👍

lukelloyd1985 commented:
@nvuillam, how come the PR got closed rather than merged? Did I do something wrong?

nvuillam commented:
@lukelloyd1985 I played with alpha to solve CI issues (which are still not solved, damn docker !)

So I deleted and recreated (then merged alpha), I think your PR has been closed when I deleted alpha branch to recreate it , sorry you can reopen it ! :)

nvuillam commented:
reopen

lukelloyd1985 commented:
> @lukelloyd1985 I played with alpha to solve CI issues (which are still not solved, damn docker !)
> 
> So I deleted and recreated (then merged alpha), I think your PR has been closed when I deleted alpha branch to recreate it , sorry you can reopen it ! :)

Need a new alpha branch by the looks of it :)

nvuillam commented:
@lukelloyd1985 created :)
(can't promise it won't move, the CI issues are still not solved :'( :'( )
